core/schema/host.go

```diff
@@ -48,27 +46,10 @@ type setSecretFileArgs struct {
 	Path string
 }
 
-func (s *hostSchema) setSecretFile(ctx *router.Context, _ any, args setSecretFileArgs) (*core.Secret, error) {
-	if s.host.DisableRW {
-		return nil, core.ErrHostRWDisabled
-	}
-
-	var absPath string
-	var err error
-	if filepath.IsAbs(args.Path) {
-		absPath = args.Path
-	} else {
-		absPath = filepath.Join(s.host.Workdir, args.Path)
-	}
-
-	absPath, err = filepath.EvalSymlinks(absPath)
-	if err != nil {
-		return nil, fmt.Errorf("eval symlinks: %w", err)
-	}
-
-	secretFileContent, err := os.ReadFile(absPath)
+func (s *hostSchema) setSecretFile(ctx *core.Context, _ any, args setSecretFileArgs) (*core.Secret, error) {
+	secretFileContent, err := s.bk.ReadCallerHostFile(ctx, args.Path)
```

Comment:
@grouville @helderco Just FYI was able to re-integrate support for this API. Ended being able to add this `ReadCallerHostFile` util that lets the engine server read files from the client caller directly into memory, so preserves all the same security properties as before.

Reply:
That's great! 🚀